### PR TITLE
Filter textDocument/didChangeWatchedFiles events according to kind

### LINKED ISSUE #938: didChangeWatchedFiles "kind" registration option not respected
**Describe the bug**
I registered for only "change" events, but my server gets sent "create" and "delete" events as well. 

**To Reproduce**
When registering for watched file events, set `DidChangeWatchedFilesRegistrationOptions.Kind` to "2" (which is `Change` only). Notice Emacs still sends "create" and "delete" notifications.

**Expected behavior**
Emacs only sends "change" notifications.

**Which Language Server did you use**
gopls (doesn't support watching files yet)

**OS**
Mac

### PR DESCRIPTION
**Pull request by leungbk**
Closes #938.

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -928,7 +928,7 @@ must be used for handling a particular message.")
 (defconst lsp--file-change-type
   `((created . 1)
     (changed . 2)
-    (deleted . 3)))
+    (deleted . 4)))
```
Comment (muirdm) [inline lsp-mode.el:931]: This should be 3 according to https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles, so I don't think you can change it to 4 to allow bitwise operations.

Comment (leungbk) [reply lsp-mode.el]: Oh, I didn't see that part of the spec. I mistook `WatchFile` events (which proceed in 1, 2, 4 order) for these events.

File: lsp-mode.el [hunk 2]
```diff
@@ -3256,14 +3264,16 @@ disappearing, unset all the variables related to it."
                            lsp--registered-capability-options
                            (lsp:did-change-watched-files-registration-options-watchers)
                            (seq-find
-                            (-lambda ((&FileSystemWatcher :glob-pattern))
-                              (-let [glob-regex (lsp-glob-to-regexp glob-pattern)]
-                                (or (string-match glob-regex changed-file)
-                                    (string-match glob-regex (f-relative changed-file root-folder)))))))))))
+                            (-lambda ((&FileSystemWatcher :glob-pattern :kind?))
+                              (and (or (null kind?)
```
Comment (yyoncho) [inline lsp-mode.el:3268]: this `and` is `when`?

File: lsp-mode.el [hunk 3]
```diff
@@ -3256,14 +3264,16 @@ disappearing, unset all the variables related to it."
...
+                                       (> (logand kind? watch-bit) 0))
```
Comment (yyoncho) [inline lsp-mode.el:3269]: nit: you may use natnump instead of `> 0`

File: lsp-mode.el (hunk 2 above)
Comment (leungbk) [reply lsp-mode.el]: Done.

File: lsp-mode.el (hunk 3 above)
Comment (leungbk) [reply lsp-mode.el]: I don't think `natnump` works here since we want to reject 0 as well, and 0 is a natural number.

**Pull request merged**